thirtyinch/src/test/java/net/grandcentrix/thirtyinch/TiPresenterTest.java

```diff
@@ -37,6 +39,104 @@
 
     private TiView mView;
 
+    @Test
+    public void attachDifferentView() throws Exception {
+        TiView viewOverride = mock(TiView.class);
+        mPresenter.create();
+
+        mPresenter.attachView(mView);
+        assertThat(mPresenter.getView(), equalTo(mView));
+
+        try {
+            mPresenter.attachView(viewOverride);
+            fail("no exception thrown");
+        } catch (IllegalStateException e) {
+            assertTrue(e.getMessage().contains("detachView"));
+        }
+    }
+
+    @Test
+    public void attachNullView() throws Exception {
+
+        mPresenter.create();
+        try {
+            mPresenter.attachView(null);
+            fail("no exception thrown");
+        } catch (IllegalStateException e) {
+            assertTrue(e.getMessage().contains("detachView()"));
+        }
+    }
+
+    @Test
+    public void attachSameViewTwice() throws Exception {
+        mPresenter.create();
+
+        mPresenter.attachView(mView);
+        assertThat(mPresenter.getView(), equalTo(mView));
+
+        mPresenter.attachView(mView);
+        assertThat(mPresenter.getView(), equalTo(mView));
+    }
+
+    @Test
+    public void attachViewToDestroyedPresenter() throws Exception {
+        mPresenter.create();
+        mPresenter.destroy();
+
+        try {
+            mPresenter.attachView(mView);
+            fail("no exception thrown");
+        } catch (IllegalStateException e) {
+            assertTrue(e.getMessage().contains("terminal state"));
+        }
+    }
+
+    @Test
+    public void attachWithoutInitialize() throws Exception {
+
+        // not calling
+        // mPresenter.create();
```

Comment:
Can be removed. The method should say what happen here.
Maybe a better name is `attachWithoutOnCreate`. But I'm also fine with the current name